### PR TITLE
Let stacktrace logging of exceptions inside MDG runner obey gradle --stacktrace option

### PR DESCRIPTION
**Pull request by lukebemish**
Requires #31 due to touching some of the same bits of code; marking as a draft till that's merged and I can rebase

### CODE REVIEW

File: gradle-plugin/src/main/groovy/org/groovymc/modsdotgroovy/gradle/ModsDotGroovyGradlePlugin.groovy [hunk 1]
```diff
@@ -38,6 +39,7 @@ abstract class ModsDotGroovyGradlePlugin implements Plugin<Project> {
         project.getGradle().getSharedServices().registerIfAbsent(ConvertService.name, ConvertService) { BuildServiceSpec<ConvertService.Parameters> it ->
             it.parameters.threads.set(propertyOf(project, ConvertService.THREAD_COUNT_PROPERTY).orElse("4"))
             it.parameters.logLevel.set(propertyOf(project, ConvertService.LOG_LEVEL_PROPERTY).orElse(closestLogLevel(project.gradle.startParameter.logLevel)))
+            it.parameters.hideStacktrace.set(propertyOf(project, ConvertService.STACKTRACE_PROPERTY).map { it.toBoolean() }.orElse(project.gradle.startParameter.showStacktrace === ShowStacktrace.INTERNAL_EXCEPTIONS))
```
Comment (PaintNinja) [inline gradle-plugin/src/main/groovy/org/groovymc/modsdotgroovy/gradle/ModsDotGroovyGradlePlugin.groovy]: Similar to my comment in another PR, please use a method reference if possible

File: stock-plugins/forgelike/src/main/groovy/org/groovymc/modsdotgroovy/plugin/forgelike/ForgeLikePlugin.groovy [hunk 2]
```diff
@@ -241,7 +241,7 @@ final class ForgeLikePlugin extends ModsDotGroovyPlugin {
         }
 
         if (this.onlineBehavior === ConversionSettings.OnlineBehavior.ERROR) {
-            throw new PluginResult.MDGPluginException('Attempted to infer updateJSONURL for modId "${modId}", but running with onlineBehavior=ERROR.')
+            throw new PluginResult.MDGPluginException("Attempted to infer updateJSONURL for mod ${modId}, but running with onlineBehavior=ERROR.")
```
Comment (PaintNinja) [inline stock-plugins/forgelike/src/main/groovy/org/groovymc/modsdotgroovy/plugin/forgelike/ForgeLikePlugin.groovy]: Missing "" around modId
```suggestion
            throw new PluginResult.MDGPluginException("Attempted to infer updateJSONURL for mod \"${modId}\", but running with onlineBehavior=ERROR.")
```

File: runner/src/main/groovy/org/groovymc/modsdotgroovy/runner/ModsDotGroovyRunner.groovy [hunk 3]
```diff
@@ -23,13 +23,26 @@ class ModsDotGroovyRunner implements AutoCloseable {
         optimizationOptions['indy'] = true
     }
 
+    private static boolean STACKTRACE = !Boolean.getBoolean("org.groovymc.modsdotgroovy.conversion.hidestacktrace")
```
Comment (PaintNinja) [inline runner/src/main/groovy/org/groovymc/modsdotgroovy/runner/ModsDotGroovyRunner.groovy]:
```suggestion
    private static final boolean STACKTRACE = !Boolean.getBoolean("org.groovymc.modsdotgroovy.conversion.hidestacktrace")
```

**Review by PaintNinja — CHANGES_REQUESTED** [review]

File: gradle-plugin/src/main/groovy/org/groovymc/modsdotgroovy/gradle/ModsDotGroovyGradlePlugin.groovy (hunk 1 above)
Comment (lukebemish) [reply gradle-plugin/src/main/groovy/org/groovymc/modsdotgroovy/gradle/ModsDotGroovyGradlePlugin.groovy]: Still unconvinced that this is a good idea due to the decreased readability and exposure of the extension method class, but I don't care enough to argue so I'll change that.

**Pull request merged**